### ISSUE TITLE
Error installing package

### ISSUE THREAD

**Issue by msurasky** [gbrian/grafana-simple-sql-datasource#8]
When issuing the npm install command on my Linux box (Linux Mint 18.2 Cinnamon 64bit) I get this error. Any ideas?

```
> contextify@0.1.15 install /home/msurasky/Documents/grafana-simple-sql-datasource-master/node_modules/contextify
> node-gyp rebuild

/bin/sh: 1: node: not found
gyp: Call to 'node -e "require('nan')"' returned exit status 127 while in binding.gyp. while trying to load binding.gyp
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/share/node-gyp/lib/configure.js:354:16)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Linux 4.8.0-53-generic
gyp ERR! command "/usr/bin/nodejs" "/usr/bin/node-gyp" "rebuild"
gyp ERR! cwd /home/msurasky/Documents/grafana-simple-sql-datasource-master/node_modules/contextify
gyp ERR! node -v v4.2.6
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 
npm ERR! Linux 4.8.0-53-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "install"
npm ERR! node v4.2.6
npm ERR! npm  v3.5.2
npm ERR! code ELIFECYCLE

npm ERR! contextify@0.1.15 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the contextify@0.1.15 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the contextify package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs contextify
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls contextify
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/msurasky/Documents/grafana-simple-sql-datasource-master/npm-debug.log
```

**Comment by gbrian** [comment]
Hi @msurasky ,

Got a comment about the need of having Python installed. 
I'm not sure which dependency requieres it I guess mssql module.
Please install Python an let me know if it worked.

Thanks

**Comment by msurasky** [comment]
Hi @gbrian!

Thanks for replying back.

I actually decided to implement the proxy on a Linux Mint OS. The process was much smoother there. I used [this blog post](http://blog.teamtreehouse.com/install-node-js-npm-linux) in order to install Node and NPM and it all worked more or less just fine.

The only road block I stumble upon was an error caused by GYP when trying to start the proxy. I don't think I have the logs for that... I'm sorry (but I believe it was something like "gyp ERR! stack Error: `make` failed with exit code: 2"). 

I managed to fix that error by using a newer version of jsdom (the package.json in source code requires 3.1.2 and I changed it with 11.1.0). I was told that newer versions of jsdom do not require GYP so apparently that is the reason why this is now working fine.

The other thing that I found difficult to understand (especially being a Grafana newbie) is that the files had to be placed on the Grafana box themselves (in addition to the Proxy) for the data source type to become available. Even after testing the proxy ([success]) I spent an hour trying to find out how the heck I could create the data source (I knew the URL was fine because otherwise the test would not have passed, the problem was I couldn't understand how to make the data source type for SimpleSql available.

I would encourage you to invest more time in the documentation. I really think you have a very neat agent here, but the difficulty of implementation (when little to no documentation is available) may throw some people off. Don't take this comment as disparaging, all the contrary I think you have a marvelous plugin and I would love to see a bigger community of people implementing it, that's all.

Again, thanks for your help and for a wonderful Grafana Agent. I hope I can get to learn to do more stuff and get tons of metrics out of my SQL Server box!

Cheers!
